modules/dnn/src/layers/slice_layer.cpp

```diff
@@ -205,44 +205,43 @@ class SliceLayerImpl : public SliceLayer
         std::vector<UMat> inputs;
         std::vector<UMat> outputs;
 
-        bool use_half = (inputs_.depth() == CV_16S);
         inputs_.getUMatVector(inputs);
         outputs_.getUMatVector(outputs);
 
-        if (inputs[0].dims < 4 || (total(shape(outputs[0]), 0, 2) % 4 != 0) ||
-            (total(shape(outputs[0]), 2) % 4 != 0))
+        CV_Assert(outputs.size() == finalSliceRanges.size());
+
+        if (inputs[0].dims != 4)
             return false;
 
-        String opts;
-        if (use_half)
-            opts = "-DDtype=half -DDtype4=half4 -DDtype8=half8";
-        else
-            opts = "-DDtype=float -DDtype4=float4 -DDtype8=float8";
-        const UMat& inpMat = inputs[0];
+        const UMat& input = inputs[0];
+        const MatShape inputShape = shape(input);
+        const size_t inputPlaneSz = total(inputShape, 2);
         for (size_t i = 0; i < outputs.size(); i++)
         {
-            int groups = outputs[i].size[0];
-            int channels = outputs[i].size[1];
-            int rows = outputs[i].size[2];
-            int cols = outputs[i].size[3];
-
-            ocl::Kernel kernel("slice", ocl::dnn::slice_oclsrc, opts);
-            size_t local[] = { 128 };
-            size_t global[] = { (size_t)groups * channels / 4 * local[0] };
-            int idx = 0;
-            kernel.set(idx++, ocl::KernelArg::PtrReadOnly(inpMat));
-            kernel.set(idx++, (int)(inpMat.size[2] * inpMat.size[3]));
-            kernel.set(idx++, (int)(rows * cols));
-            kernel.set(idx++, (int)inpMat.size[3]);
-            kernel.set(idx++, (int)cols);
-            kernel.set(idx++, (int)finalSliceRanges[i][2].start);
-            kernel.set(idx++, (int)finalSliceRanges[i][3].start);
-            kernel.set(idx++, ocl::KernelArg::PtrWriteOnly(outputs[i]));
-            bool ret = kernel.run(1, global, local, false);
-            if (!ret)
+            MatShape outputShape = shape(outputs[i]);
+            if (total(outputShape, 2) != inputPlaneSz && outputShape[1] != 1)
                 return false;
         }
 
+        for (size_t i = 0; i < outputs.size(); i++)
+        {
+            UMat& output = outputs[i];
+#if 0  // TODO: core - support non-contiguous ND layouts
+            input(finalSliceRanges[i]).copyTo(output);
```

Comment:
Seems strange why input is not contiguous.

Reply:
Input is contiguous, but ROI/slice is not.